### PR TITLE
CXP-1294: Endpoint GET mapped-products

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            'code' => 'description',
+            'type' => 'pim_catalog_text',
+            'scopable' => true,
+            'localizable' => false,
+        ]);
+
+        $this->createChannel('print', ['en_US', 'fr_FR']);
+
+        $productBlue = $this->createProduct(Uuid::fromString('8985de43-08bc-484d-aee0-4489a56ba02d'), [
+//            new SetTextValue('name', 'ecommerce', 'en_US', 'Blue print'),
```
Comment (eoliveauakeneo) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php:59]: This line should be removed if not used

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            new SetTextValue('name', 'print', 'en_US', 'Blue print'),
+            new SetTextValue('description', 'print', null, 'Blue description'),
+        ]);
+        $productGreen = $this->createProduct(Uuid::fromString('00380587-3893-46e6-a8c2-8fee6404cc9e'), [
+//            new SetTextValue('name', 'ecommerce', 'en_US', 'Green print'),
```
Comment (eoliveauakeneo) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php:64]: This line should be removed if not used

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetProductUuidsAction.php [hunk 3]
```diff
@@ -133,15 +133,15 @@ private function paginate(Catalog $catalog, array $uuids, ?string $searchAfter,
                         'limit' => $limit,
                         'updated_after' => $updatedAfter,
                         'updated_before' => $updatedBefore,
-                    ]),
+                    ], RouterInterface::ABSOLUTE_URL),
```
Comment (willy-ahva) [inline components/catalogs/back/src/Infrastructure/Controller/Public/GetProductUuidsAction.php:136]: could you explain pls ?

File: components/catalogs/back/tests/Acceptance/ApiContext.php [hunk 4]
```diff
@@ -685,4 +688,202 @@ public function theCatalogProductMappingSchemaShouldBeEmptyInThePim(): void
...
+        ));
+
+        // add product mapping to catalog
+        $this->setCatalogProductMapping('db1079b6-f397-4a6a-bae4-8658e64ad47c', [
+            'uuid' => [
+                'source' => 'uuid',
+                'scope' => null,
+                'locale' => null,
+            ],
+            'title' => [
```
Comment (willy-ahva) [inline components/catalogs/back/tests/Acceptance/ApiContext.php:755]: please add a target which will not be match by the product to test `"target_name": null`

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                'locale' => 'en_US',
+            ],
+            'short_description' => [
+                'source' => 'description',
+                'scope' => 'print',
+                'locale' => null,
+            ],
+        ]);
+
+        $this->client->request(
```
Comment (RiffVentura) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetMappedProductsActionTest.php:101]: IMO it should have another product and test that page 2 behaves as expected to truly test pagination

File: components/catalogs/back/src/Application/Handler/GetProductsHandler.php [hunk 6]
```diff
@@ -6,20 +6,20 @@
...
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  *
- * @phpstan-import-type Product from GetProductsQueryInterface
+ * @phpstan-import-type Product from GetProductsWithFilteredValuesQueryInterface
  */
 final class GetProductsHandler
 {
     public function __construct(
-        private GetProductsQueryInterface $getProductsQuery,
+        private GetProductsWithFilteredValuesQueryInterface $getProductsQuery,
```
Comment (RiffVentura) [inline components/catalogs/back/src/Application/Handler/GetProductsHandler.php:22]:
```suggestion
        private GetProductsWithFilteredValuesQueryInterface $getProductsWithFilteredValuesQuery,
```

**Review by RiffVentura — APPROVED** [review]

**Comment by RiffVentura** [comment]
the small renaming might need to be done to `GetProductQuery` as well since it returns one value-filtered product

File: components/catalogs/back/src/Application/Handler/GetMappedProductsHandler.php [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            function (array $product) use ($productMappingSchema, $productMapping): array {
+                $mappedProduct = [];
+
+                /** @var string $key */
+                foreach (\array_keys($productMappingSchema['properties']) as $key) {
+                    $sourceValue = '';
+
+                    if ('uuid' === $key) {
+                        $sourceValue = $product['uuid'];
+                    } elseif (\array_key_exists($key, $productMapping)) {
```
[the remark this answers is not in the document]
Comment (masacc) [reply components/catalogs/back/src/Application/Handler/GetMappedProductsHandler.php]: Indeed, this is said here : https://www.notion.so/akeneo/PDR-2-preparation-Mapping-for-Apps-bd79e0e3c2044e9a891c0c44f00b7f37?p=e5d2798cc01749ef86c0133eb816d970&pm=s

@guillaumefer-akeneo @eoliveauakeneo @willy-ahva do you have created a ticket to handle this issue ?

**Comment by masacc** [comment]
> the small renaming might need to be done to `GetProductQuery` as well since it returns one value-filtered product

We planed to do this when the endpoint `/catalog/{id}/mapped-products/{uuid}` will be developed.
